blocks/components/editable/index.js

```diff
@@ -349,6 +350,8 @@ export default class Editable extends wp.element.Component {
 		this.setState( {
 			formats: merge( {}, this.state.formats, formats )
 		} );
+
+		this.editor.setDirty( true );
```

Comment:
On Chrome? I confirmed that it seems to work on Chrome too. I'm not sure why because if you throw a log statement in after all the formatting is set then this.editor.isDirty() is false which means in onChange it shouldn't save, however it is saving... One of the other developers here suggested that it might be picking up a node change event and setting the dirty flag later.

On Firefox there are a few problems. Firstly it seems to always throw an IndexSizeError within the call to ```this.editor.formatter.apply( format )``` (apparently it is trying to restore the original selection but the text node that it is trying to select has since been split by the formatting operation and the previous range is not valid). I haven't attempted to fix this in my pull request because it seems out of scope. If I do wrap these problematic calls with a try/catch then as with Chrome the isDirty function still returns false after the formatting has been applied but unlike Chrome it never seems to get corrected so it does not save in the onChange handler.

So the reason I put this.editor.setDirty(true) there is so that this will work when the Firefox IndexSizeError problem has been fixed. Still I suppose that really we just need to get it fixed in TinyMCE.